adapters/oidc/js/src/main/resources/keycloak.d.ts

```diff
@@ -215,6 +215,8 @@ declare namespace Keycloak {
 		redirectUri?: string;
 	}
 
+	interface KeycloakRegisterOptions extends Omit<KeycloakLoginOptions, 'action'> { }

```

Comment:
I don't think this is quite right.  Today it is valid to pass the action as a register option as long as it is set to 'register'.  So this could possibly break existing code.

What do you think of this instead?
`type KeycloakRegisterOptions = Omit<KeycloakLoginOptions, 'action'> & {action?: 'register'}`

Reply:
@jonkoops ^^ ?

Reply:
It might be valid, but in code we never use the `action` property for any registration related methods. If anyone is passing in this parameter, or any other parameter not in the interface for that matter it will rightly produce an exception during compilation of TypeScript code.

The question is, is that really a breaking change? It does not affect any code running in production, rather only projects taking advantage of TypeScript where the author makes a conscious decision to update this specific dependency and build a new version of their application.

I think it's good to be explicit about what is present and used and what is not to avoid confusion, and I think that others using TypeScript would want to have the same behaviour.

That said, if we do want to keep scope around we might as well use `KeycloakLoginOptions` directly as the `action` property is already optional there.

Reply:
I do think it really is a breaking change.

But I can buy the argument that we should discourage anyone from passing in the `action` property anyway.  So maybe that's not such a bad thing.

I'll approve as-is.